### ISSUE TITLE
Automatically test discoverable KMS

### ISSUE THREAD

**Issue by jimmykarily** [kairos-io/kairos#2182]
We introduced a test here: https://github.com/kairos-io/kcrypt-challenger/pull/45/files#diff-7b5de4f9b392e3c469cc56719eb808add9da3a962994c0a891f761a8fc57a6be

but it doesn't work so we skipped it to unblock the release. We should make it work and enable it.

**Comment by ci-robbot** [comment]
Triage note: still open, but the breadcrumbs are stale enough that whoever picks this up will lose an afternoon just finding them.

**The referenced PR.** `kcrypt-challenger#45` in the link now 404s. The repo was renamed to `kcrypt-discovery-challenger` and then archived when it was folded into this monorepo. The PR lives at kairos-io/kcrypt-discovery-challenger#45, and the pending spec it introduced was:

- `tests/encryption_test.go`, `XWhen("discovering KMS with mdns", Label("discoverable-kms"), ...)` (the `X` is Ginkgo Pending)
- Helper: `deploySimpleMDNSServer`, which spins up a second QEMU VM running [kairos-io/simple-mdns-server](https://github.com/kairos-io/simple-mdns-server) that advertises `discoverable-kms.local`

**Why it was skipped.** The in-code TODO records it: _"Use bridge networking because the default qemu networking won't cut it. The mdns response can't reach the VM."_ Default user-mode QEMU networking does not forward the multicast the test relies on. That is the real blocker, and it did not go away with the migration.

**What the monorepo has today.**
- The feature under test still exists in-tree at `kcrypt/discovery/client/mdns.go` (`MDNSServiceType = "_kcrypt._tcp"`, `queryMDNS`), unchanged, so the test target is real.
- `tests/encryption_test.go` has the migrated remote-KMS specs (`encryption-remote-auto`, `encryption-remote-static`, `encryption-remote-https-*`) but not the mDNS one, and no `deploySimpleMDNSServer` helper came across. That work still has to happen.
- `kairos-io/simple-mdns-server` is now archived too, so reviving the test also means either vendoring that helper image or replacing it with something advertising `_kcrypt._tcp` from an adjacent VM.

**To close this issue** someone needs to: port the `XWhen` block and helper into `tests/encryption_test.go`, wire a QEMU network mode that actually carries the mDNS response (bridge or a two-VM socket network), and add a `discoverable-kms` label to whichever leg of `.github/workflows/reusable-qemu-test.yaml` should run it.

Not proposing to close, the ask is still valid, just recording where things stand so the next person does not chase a dead link.
